src/main/java/org/wildfly/security/auth/server/RealmIdentity.java

```diff
@@ -207,5 +244,9 @@ public boolean verifyEvidence(final Evidence evidence) throws RealmUnavailableEx
         public boolean exists() throws RealmUnavailableException {
             return false;
         }
+
+        public SecurityRealm getSecurityRealm() {
```

Comment:
This is the part I don't like about this: exposing the realm seems like the abstraction is leaking a little bit.  Do you think there could be a way to do this without requiring that the realm to be exposed by the identity?